### PR TITLE
Update configuring-openid-connect-in-amazon-web-services.md

### LINKED ISSUE #22239: Update configuring-openid-connect-in-amazon-web-services.md
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#configuring-the-role-and-trust-policy

### What part(s) of the article would you like to see updated?

I would like to make the AWS IAM role trust policies correct. They are currently incorrect.

### Additional information

Please take a look at my pull request.

### PR DESCRIPTION
**Pull request by liwadman**
Removed use of non-existent condition key (:iss claim), removed incorrect use of ForAllValues, changed documentation explanation to be correct.

Source: Work on the AWS Identity team

### Why:

Closes [[issue link]](https://github.com/github/docs/issues/22239)

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Removes the non-existent AWS condition keys (:iss), removes the incorrect use of ForAllValues, changes the rightup to reflect what the example actually does and how it works.

### Check off the following:

- [ X] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [ X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@liwadman Thanks for opening a PR! I've got it triaged for review ⚡

**Review by doctorhamed — COMMENTED** [review]
Drhamedhiasat25@gmail.com

File: content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ Edit the trust policy to add the `sub` field to the validation conditions. For e
 }
 ```
 
-In the following example, `ForAllValues` is used to match on multiple condition keys, and `StringLike` is used to match any ref in the specified repository. Note that `ForAllValues` is [overly permissive](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html) and should not be used on its own in an `Allow` effect. For this example, the inclusion of `StringLike` means that an empty set in `ForAllValues` will still not pass the condition:
+In the following example, StringLike with a wildcard operator, `*`, is used to allow any branch or enviroment from the `octo-org/octo-repo` github organization and repo to assume a role in AWS.
```
Comment (kchandra548) [inline content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md:52]: This also allows the PullRequest ref.

File: content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md [hunk 2]
```diff
@@ -65,8 +65,7 @@ In the following example, `ForAllValues` is used to match on multiple condition
                 "StringLike": {
                     "token.actions.githubusercontent.com:sub": "repo:octo-org/octo-repo:*"
                 },
-                "ForAllValues:StringEquals": {
-                    "token.actions.githubusercontent.com:iss": "https://token.actions.githubusercontent.com",
```
Comment (kchandra548) [inline content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md:69]: Nice observation 👍 , yeah we don't need condition on `iss` claim

File: content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md (hunk 1 above)
Comment (martin389) [reply content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md:52]: @kchandra548 -- what do you think of this suggestion?

```suggestion
In the following example, `StringLike` is used with a wildcard operator (`*`) to allow any branch, pull request merge branch, or environment from the `octo-org/octo-repo` organization and repository to assume a role in AWS. 
```

**Review by kchandra548 — DISMISSED** [review]
LGTM!!

**Comment by martin389** [comment]
Thanks for this PR! I've added a commit based on engineering's feedback, and they've now approved it.

**Review by martin389 — APPROVED** [review]
Changes look good in staging.

**Comment by Boesmanbekkie** [comment]
> ## Automatically generated comment ℹ️
> **This comment is automatically generated and will be overwritten every time changes are committed to this branch.**
> 
> The table contains an overview of files in the `content` directory that have been changed in this pull request. It's provided to make it easy to review your changes on the staging site. Please note that changes to the `data` directory will not show up in this table.
> 
> ### Content directory changes
> _You may find it useful to copy this table into the pull request summary. There you can edit it to share links to important articles or changes and to give a high-level overview of how the changes in your pull request support the overall goals of the pull request._
> 
> **Source**	**Preview**	**Production**	**What Changed**
> [`actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services.md`](https://github.com/github/docs/blob/9420d8614377b9343bbe4af39a2b46e26e22a915/content%2Factions%2Fdeployment%2Fsecurity-hardening-your-deployments%2Fconfiguring-openid-connect-in-amazon-web-services.md)	[fpt](https://docs-22240-33950b.preview.ghdocs.com/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> [ghec](https://docs-22240-33950b.preview.ghdocs.com/enterprise-cloud@latest/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> ghes@ [3.7](https://docs-22240-33950b.preview.ghdocs.com/enterprise-server@3.7/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) [3.6](https://docs-22240-33950b.preview.ghdocs.com/enterprise-server@3.6/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) [3.5](https://docs-22240-33950b.preview.ghdocs.com/enterprise-server@3.5/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> [fpt](https://docs.github.com/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> [ghec](https://docs.github.com/enterprise-cloud@latest/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> ghes@ [3.7](https://docs.github.com/enterprise-server@3.7/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) [3.6](https://docs.github.com/enterprise-server@3.6/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) [3.5](https://docs.github.com/enterprise-server@3.5/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services)
> fpt: Free, Pro, Team ghec: GitHub Enterprise Cloud ghes: GitHub Enterprise Server ghae: GitHub AE

**Pull request merged**